crates/weaver_forge/src/debug.rs

```diff
@@ -111,7 +110,10 @@ mod tests {
 
     #[test]
     fn test_print_dedup_errors() {
-        let logger = weaver_common::TestLogger::new();
+        // Setup TestLog to capture log messages and keep a reference to it
+        let test_log = weaver_common::TestLog::new();
+        test_log.clone().init().expect("Failed to init logger");
```

Comment:
This split between the constructor and the init method seems really error-prone to me. Couldn’t we have a constructor that performs the initialization? Otherwise, there’s a high chance someone will misuse the logger.

Reply:
Fixed in #727